src/app/games/awale/tests/AwaleRules.spec.ts

```diff
@@ -19,33 +20,87 @@ describe('AwaleRules', () => {
             new AwaleMinimax(rules, 'AwaleMinimax'),
         ];
     });
-    it('should capture', () => {
-        const board: number[][] = [
+    it('should distribute', () => {
+        // Given a state where the player can perform a distributing move
+        const board: Table<number> = [
+            [0, 0, 0, 0, 3, 4],
+            [0, 0, 0, 0, 0, 0],
+        ];
+        const state: AwaleState = new AwaleState(board, 0, [0, 0]);
+        // When performing a distribution
+        const move: AwaleMove = AwaleMove.FIVE;
+        // Then the distribution should be performed as expected
+        const expectedBoard: Table<number> = [
+            [0, 0, 0, 0, 3, 0],
+            [0, 0, 1, 1, 1, 1],
+        ];
+        const expectedState: AwaleState = new AwaleState(expectedBoard, 1, [0, 0]);
+        RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+    });
+    it('should not drop a piece in the starting space', () => {
+        // Given a state where the player can perform a distributing move with at least 12 stones
+        const board: Table<number> = [
+            [0, 0, 0, 0, 0, 18],
+            [0, 0, 0, 0, 0, 0],
+        ];
+        const state: AwaleState = new AwaleState(board, 0, [0, 0]);
+        // When performing a distribution
+        const move: AwaleMove = AwaleMove.FIVE;
+        // Then the distribution should be performed as expected, and leave 0 stones in the starting space
+        const expectedBoard: Table<number> = [
+            [2, 1, 1, 1, 1, 0],
+            [2, 2, 2, 2, 2, 2],
+        ];
+        const expectedState: AwaleState = new AwaleState(expectedBoard, 1, [0, 0]);
+        RulesUtils.expectMoveSuccess(rules, state, move, expectedState);
+    });
+    it('should capture for player zero', () => {
+        // Given a state where a capture is possible for player 0
+        const board: Table<number> = [
             [0, 0, 0, 0, 1, 1],
             [0, 0, 0, 0, 1, 1],
         ];
-        const expectedBoard: number[][] = [
+        const state: AwaleState = new AwaleState(board, 0, [1, 2]);
+        // When performing a move that will capture 3+2 seeds
```

Comment:
does not capture 3 + 2 seed, there is not even 5 capturables seeds left!

Reply:
copypasta :x  corrigé